packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -249,6 +251,22 @@ export class MetaMaskHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  public async ethSwitchChain(chainId: number): Promise<void> {
+    const hexChainId = ethers.utils.hexValue(chainId);
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
+    try {
+      // at this point, we know that we're in the context of a valid MetaMask provider
+      await provider.request({ method: "wallet_switchEthereumChain", params: [{ chainId: hexChainId }] });
+    } catch (e: any) {
```

Comment:
Since adding a chain if not added in MetaMask yet is a new feature (which looks like error handling here), I'll use MetaMask's type guard in the follow-up PR that will add the (C- in our case) chain if missing